File: gin-go-backend/internal/errors/http_errors.go
package errors

import "gin-go-backend/internal/types"

func ToProblem(appErr AppError, instance, requestID, traceID, spanID string) types.ProblemDetails {
	return types.ProblemDetails{
		Type: appErr.Type, Title: appErr.Title, Status: appErr.Status, Detail: appErr.Detail,
		Instance: instance, RequestID: requestID, TraceID: traceID, SpanID: spanID,
	}
}

File: actix-rust-backend/src/errors/http_errors.rs
use serde::Serialize;
use utoipa::ToSchema;

#[derive(Debug, Serialize, ToSchema)]
pub struct ProblemDetails {
    #[serde(rename = "type")]
    pub kind: String,
    pub title: String,
    pub status: u16,
    pub detail: String,
    pub instance: String,
    #[serde(rename = "traceId")]
    pub trace_id: String,
}
